### DISCUSSION TITLE
dynamic routes and static build

### DISCUSSION THREAD

**Discussion by rossi-jeff** [vercel/next.js#48639] [Help]
### Summary

Hey Community,

I did some searching for the answer but did not find a sufficient response so forgive me if I did not RTFM

I am building the 3rd iteration of an app, afteer having done so in svelte kit and nuxt

I intend to host the application on my cheap web host by putting the build output in a folder on my host

I have pages such as:

`app/yacht/scores/[id]/page.tsx`

These pages work as expected when using `npm run dev` but after running `npm run build` and `serve build/` the return a 404 with a url ending in .txt

code can be found here https://github.com/rossi-jeff/games-next

**Comment by rossi-jeff** [comment]
would I be better off foregoing the app directory and making it work in pages?

**Comment by reyamir** [comment]
add `prefetch={false}` to `<Link>` can temporary fix this issue, but if you use `router.push`, there no way

**Reply by rossi-jeff** [reply]
hey thanks, I will try that

**Reply by rossi-jeff** [reply]
it got rid of the .txt in the url, but I still get a 404

**Comment by reyamir** [comment]
I don't know why, but {forceOptimisticNavigation: true} to router.push is fixed this issue

**Comment by rossi-jeff** [comment]
am I missing any options in here?
```
/** @type {import('next').NextConfig} */
const nextConfig = {
	experimental: {
		appDir: true,
	},
	output: 'export',
	distDir: 'build',
}

module.exports = nextConfig

```

**Reply by rossi-jeff** [reply]
there are files generated by build in `build/_next/static/chunks/app/yacht/scores/[id]/..` and the like

**Reply by rossi-jeff** [reply]
the built files are never called in the log from serve, how does one go about making a bug report?
